### PR TITLE
fix: Add ERC20 decimals query when calculating fees

### LINKED ISSUE #29: For the PriceFeed all tokens have 18 decimals
## Summary of Bug
In the function CheckFeeThreshold the exponent is hardcoded, meaning that if we have tokens that do not have 18 decimals, the calculated price will be off by a lot. That value should be queried from the token's contract, using the "decimals" attribute from the ERC20 standard. Then we can keep that value in memory because it's fixed and won't change.
Ref: https://github.com/umee-network/peggo/blob/main/orchestrator/coingecko/coingecko.go#L137

----

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by facundomedica**
This closes #29 
And also adds a cast check when getting back the results from coingecko, without it the program would crash when querying for a non-existant token address.

### CODE REVIEW

File: orchestrator/ethereum/peggy/utils.go [hunk 1]
```diff
@@ -74,20 +74,47 @@ func (s *peggyContract) GetERC20Symbol(
...
+	}
+
+	callOpts := &bind.CallOpts{
+		From:    callerAddress,
+		Context: ctx,
+	}
+
+	decimals, err = erc20Wrapper.Decimals(callOpts)
+	if err != nil {
+		err = errors.Wrap(err, "ERC20 [decimals] call failed")
```
Comment (alexanderbez) [inline orchestrator/ethereum/peggy/utils.go:115]:
```suggestion
		err = errors.Wrap(err, "ERC20 'decimals' call failed")
```

File: orchestrator/main_loops.go [hunk 2]
```diff
@@ -280,15 +281,34 @@ func (p *peggyOrchestrator) CheckFeeThreshold(
...
+		Msg("got token decimals")
+
 	tokenPriceInUSD, err := p.priceFeeder.QueryUSDPrice(erc20Contract)
 	if err != nil {
 		return false
 	}
 
 	tokenPriceInUSDDec := decimal.NewFromFloat(tokenPriceInUSD)
-	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -18).Mul(tokenPriceInUSDDec)
+	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -int32(decimals)).Mul(tokenPriceInUSDDec)
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:301]: Should we be safe and return an error on overflow? i.e. return an error if `decimals > math.MaxInt32`?

File: orchestrator/main_loops.go [hunk 3]
```diff
@@ -280,15 +281,34 @@ func (p *peggyOrchestrator) CheckFeeThreshold(
...
 	minFeeInUSDDec := decimal.NewFromFloat(minFeeInUSD)
 
+	p.logger.Debug().
+		Str("token_contract", erc20Contract.String()).
+		Float64("token_price_in_usd", tokenPriceInUSD).
+		Int64("total_fees", totalFee.Int64()).
+		Float64("total_fee_in_usd", totalFeeInUSDDec.InexactFloat64()).
+		Float64("min_fee_in_usd", minFeeInUSDDec.InexactFloat64()).
+		Msg("checking if token fees meets set threshold amount")
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:310]:
```suggestion
		Msg("token fees meet minimum batch fee threshold")
```

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: orchestrator/main_loops.go (hunk 2 above)
Comment (facundomedica) [reply orchestrator/main_loops.go:301]: Gonna add that 👌

File: orchestrator/main_loops.go (hunk 3 above)
Comment (facundomedica) [reply orchestrator/main_loops.go:310]: This is not necessarily true, the check is performed in the return

File: orchestrator/main_loops.go (hunk 2 above)
Comment (facundomedica) [reply orchestrator/main_loops.go:301]: Ah, we don't need it, decimals is a uint8 (even in the ERC20 contract is being stored as a uint8).

Comment (alexanderbez) [reply orchestrator/main_loops.go:301]: Let's add a comment then 👍

File: orchestrator/main_loops.go (hunk 3 above)
Comment (alexanderbez) [reply orchestrator/main_loops.go:310]: You're right!

**Review by alexanderbez — APPROVED** [review]

**Review by alexanderbez — APPROVED** [review]

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**